compiler/scenario-service/client/src/DA/Daml/LF/ScenarioServiceClient.hs

```diff
@@ -142,23 +145,30 @@ getNewCtx Handle{..} Context{..} = withLock hContextLock $ withSem hConcurrencyS
       (S.toList unloadPackages)
       ctxDamlLfVersion
       ctxSkipValidation
-  writeIORef hLoadedPackages newLoadedPackages
-  writeIORef hLoadedModules ctxModules
-  res <- LowLevel.updateCtx hLowLevelHandle hContextId ctxUpdate
-  case res of
-    Left err -> pure (Left err)
-    Right () -> LowLevel.cloneCtx hLowLevelHandle hContextId
+  rootCtxId <- readIORef hContextId
+  runExceptT $ do
+      clonedRootCtxId <- ExceptT $ LowLevel.cloneCtx hLowLevelHandle rootCtxId
```

Comment:
How are these cloned contexts cleaned up? Sure, on the Haskell side they are GC'ed. But, don't we need to explicitly delete them to avoid leaking resources in the scenario service?

Reply:
There is a `gcCtxs` gRPC request that we send to the server https://github.com/digital-asset/daml/blob/master/compiler/damlc/daml-ide-core/src/Development/IDE/Core/Rules/Daml.hs#L908

Reply:
Thanks for clarifying!